templates/taxi/driver_detail.html

```diff
@@ -9,18 +9,20 @@ <h1>
   <p><strong>Last name:</strong> {{ driver.last_name }}</p>
   <p><strong>License number:</strong> {{ driver.license_number }}</p>
   <p><strong>Is staff:</strong> {{ driver.is_staff }}</p>
-
   <div class="ml-3">
     <h4>Cars</h4>
 
     {% for car in driver.cars.all %}
-        <hr>
-        <p><strong>Model:</strong> {{ car.model }}</p>
-        <p><strong>Manufacturer:</strong> {{ car.manufacturer.name }}</p>
-        <p class="text-muted"><strong>Id:</strong> {{car.id}}</p>
-
+      <hr>
+      <p><strong>Model:</strong> {{ car.model }}</p>
+      <p><strong>Manufacturer:</strong> {{ car.manufacturer.name }}</p>
+      <p class="text-muted"><strong>Id:</strong> {{ car.id }}</p>
     {% empty %}
       <p>No cars!</p>
     {% endfor %}
+    <div class="btn-group" role="group">
+      <a href="{% url 'taxi:driver-update' driver.id %}" class="btn btn-secondary">Update</a>
+      <a href="{% url 'taxi:driver-delete' driver.id %}" class="btn btn-danger">Delete</a>
```

Comment:
The {% url %} template tag generates URLs dynamically using the URL name defined in your urls.py file. The parameters you pass must match the named parameters expected by the URL pattern.

In your urls.py file, I see:

`path( "drivers/<int:pk>/update/", DriverUpdateView.as_view(), name="driver-update" )`

However, in your template code, I see a different URL format:

`<a href="{% url 'taxi:driver-update' driver.id %}" class="btn btn-primary btn-sm">Update</a>`

Since your URL pattern uses <int:pk>, you should use pk= in the template tag:

`<a href="{% url 'taxi:driver-update' pk=driver.id %}" class="btn btn-primary btn-sm">Update</a>`

Ensure that the href in your template matches the URL pattern, or update the URL pattern to match the parameters used in your template.